contour/src/main/kotlin/com/squareup/contour/ContourLayout.kt

```diff
@@ -387,7 +389,8 @@ open class ContourLayout(
       it.view = this
     }
     if (addToViewGroup && parent == null) {
-      viewGroup.addView(this)
+      // Cast to ViewGroup to skirt our own public deprecation
+      (viewGroup as ViewGroup).addView(this)
```

Comment:
This makes me realize that we will also need to throw an error in `addView()` in case a `ContourLayout` is passed around in a generic way and `addView()` gets called on it.

Reply:
At that point are we being over-zealous? And how would one distinguish external (i.e. `myContourLayoutInDisguise.addView`) vs internal (i.e. `this.addView`) calls?

I think there's only so much we can do to remove this footgun, but I'm open to ideas!

Reply:
Hmmmm what do you think of adding a private function that delegates to `super.addView()`?

```kotlin
private fun addViewInternal(child: View) {
  super.addView(child)
}

@Deprecated(level = ERROR)
override fun addView(child: View?) {
  super.addView(child)
}

fun <T : View> T.layoutBy(...): T {
  ...
  if (addToViewGroup && parent == null) {
    viewGroup.addViewInternal(this)
  }
  return this
}
```

The error message thrown if a view is added using `addView()` results isn't very obvious and I think we could help users avoid it this way.

Reply:
We can add the throwing for `addView`, but I wouldn't want to do it for the overloads, as they get called directly by `ViewGroup`'s implementation

Reply:
What do you think of throwing an error in the last `addView()` call if the layoutParam != LayoutSpec?

Reply:
That might just do it, and then we don't really need `addViewInternal`

Reply:
LMK how this looks